### PR TITLE
Add LieGroup Support for AutoDiffFunction

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: theseus/core/tests/test_cost_function.py [hunk 1]
```diff
@@ -185,3 +185,64 @@ def error_fn(optim_vars, aux_vars):
         weighted_error = cost_function.weighted_error()
         direct_error_computation = cost_weight_value * torch.ones(batch_size, 1)
         assert torch.allclose(weighted_error, direct_error_computation)
+
+
+def test_autodiff_cost_function_error_and_jacobians_shape_on_SO3():
```
Comment (luisenp) [inline theseus/core/tests/test_cost_function.py:190]: Do we have to test the shape (taken care already by another test) or only that the numerical value of the jacobians are correct?

**Review by luisenp — CHANGES_REQUESTED** [review]
This LGTM. I think the test can be simplified, but otherwise this looks good.

File: theseus/core/tests/test_cost_function.py (hunk 1 above)
Comment (fantaosha) [reply theseus/core/tests/test_cost_function.py:190]: Numerical values are tested in previous PR #98.

Comment (luisenp) [reply theseus/core/tests/test_cost_function.py:190]: If we already have other tests in place to check that sparse gradient computation is correct, here we should mostly test that the full combination of `AutodiffCostFunction` + `project` is put together properly and gives the expected result in a few test cases.

Comment (fantaosha) [reply theseus/core/tests/test_cost_function.py:237]: Added value tests.

**Review by mhmukadam — APPROVED** [review]

**Review by luisenp — APPROVED** [review]
